### PR TITLE
Fix native protocol inactivity timeout handling

### PR DESCRIPTION
**Pull request by NikolaiShipilov**
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by JakaMohorkoDS — APPROVED** [review]

File: shared/libraries/native_streaming_protocol/src/base_session_handler.cpp [hunk 1]
```diff
@@ -35,11 +35,11 @@ void BaseSessionHandler::startConnectionActivityMonitoring(Int heartbeatPeriod,
         connectionInactivityTimer->cancel();
         connectionInactivityTimer->expires_from_now(inactivityTimeoutMs);
         connectionInactivityTimer->async_wait(
-            [this](const boost::system::error_code& ec)
+            [errorHandler = errorHandler, session = session](const boost::system::error_code& ec)
             {
                 if (ec)
```
Comment (dejancrn) [inline shared/libraries/native_streaming_protocol/src/base_session_handler.cpp:40]: What does this mean? If operation failed we return and do nothing?

File: shared/libraries/native_streaming_protocol/src/native_streaming_client_handler.cpp [hunk 2]
```diff
@@ -236,20 +236,9 @@ void NativeStreamingClientHandler::initClientSessionHandler(SessionPtr session)
...
-                    connectionStatusChanged(ClientConnectionStatus::Reconnecting);
-                    tryReconnect();
-                });
-        }
-        else
-        {
-            connectionStatusChanged(ClientConnectionStatus::Reconnecting);
-            tryReconnect();
-        }
+
```
Comment (dejancrn) [inline shared/libraries/native_streaming_protocol/src/native_streaming_client_handler.cpp:239]: No need to close the session?

**Review by dejancrn — COMMENTED** [review]
Posted some questions just for my understaning.

File: shared/libraries/native_streaming_protocol/src/base_session_handler.cpp (hunk 1 above)
Comment (NikolaiShipilov) [reply shared/libraries/native_streaming_protocol/src/base_session_handler.cpp:42]: We do smth (here triggering connection closing) when timer expires. If timer callback called with some error code (for example timer operation cancelled on shutdown)  - yes, we return do nothing.

File: shared/libraries/native_streaming_protocol/src/native_streaming_client_handler.cpp (hunk 2 above)
Comment (NikolaiShipilov) [reply shared/libraries/native_streaming_protocol/src/native_streaming_client_handler.cpp:239]: No need to do this explicitly, as the connection will be closed automatically when the "Session" object is destroyed.

Comment (dejancrn) [reply shared/libraries/native_streaming_protocol/src/native_streaming_client_handler.cpp:239]: Ok

File: shared/libraries/native_streaming_protocol/src/base_session_handler.cpp (hunk 1 above)
Comment (dejancrn) [reply shared/libraries/native_streaming_protocol/src/base_session_handler.cpp:42]: Ok

**Review by dejancrn — APPROVED** [review]

**Pull request merged**